sdk/purview/purview-sharing-rest/test/public/shareResourcesTest.spec.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { isUnexpected, paginate, PurviewSharingClient, ShareResourceOutput } from "../../src";
+import { Recorder } from "@azure-tools/test-recorder";
+import { assert } from "chai";
+import { createClient, createRecorder } from "./utils/recordedClient";
+import { Context } from "mocha";
+
+describe("Share Resources Operations", () => {
+  let recorder: Recorder;
+  let client: PurviewSharingClient;
+
+  beforeEach(async function (this: Context) {
+    recorder = await createRecorder(this);
+    client = createClient(recorder);
+  });
+
+  afterEach(async function () {
+    await recorder.stop();
+  });
+
+  it("List all Share Resources", async function () {
+    const response = await client.path("/shareResources").get();
+    const pageData = paginate(client, response);
+
+    assert.strictEqual(response.status, "200");
+    if (isUnexpected(response)) {
+      throw response.body.error;
+    }
+
+    const result: ShareResourceOutput[] = [];
+
+    for await (const item of pageData) {
+      const shareResource = item as ShareResourceOutput;
```

Comment:
Just wondering why we need to `as ShareResourceOutput` here? My understanding is the item will be a type of `ShareResourceOutput`? Any type error received?

Reply:
Yes correct, the casting needs to happen before adding it (result) which is of type ShareResourceOutput[].

Reply:
I don't think this is expected, could you try to adjust the position for isUnexpected helper?

```typescript
if (isUnexpected(response)) {
      throw response.body.error;
    }
 const pageData = paginate(client, response);
for await (const item of pageData) {
      result.push(item );
    }

```